lib/graphql/lang/visitor.ex

```diff
@@ -0,0 +1,184 @@
+defmodule GraphQL.Lang.Visitor do
+  defmodule Node do
+    @kinds %{
+      Name: [],
+      Document: [:definitions],
+      OperationDefinition: [:name, :variableDefinitions, :directives, :selectionSet],
+      VariableDefinition: [:variable, :type, :defaultValue],
+      Variable: [:name],
+      SelectionSet: [:selections],
+      Field: [:alias, :name, :arguments, :directives, :selectionSet],
+      Argument: [:name, :value],
+      FragmentSpread: [:name, :directives],
+      InlineFragment: [:typeCondition, :directives, :selectionSet],
+      FragmentDefinition: [:name, :typeCondition, :directives, :selectionSet],
+      IntValue: [],
+      FloatValue: [],
+      StringValue: [],
+      BooleanValue: [],
+      EnumValue: [],
+      ListValue: [:values],
+      ObjectValue: [:fields],
+      ObjectField: [:name, :value],
+      Directive: [:name, :arguments],
+      NamedType: [:name],
+      ListType: [:type],
+      NonNullType: [:type],
+      ObjectTypeDefinition: [:name, :interfaces, :fields],
+      FieldDefinition: [:name, :arguments, :type],
+      InputValueDefinition: [:name, :type, :defaultValue],
+      InterfaceTypeDefinition: [:name, :fields],
+      UnionTypeDefinition: [:name, :types],
+      ScalarTypeDefinition: [:name],
+      EnumTypeDefinition: [:name, :values],
+      EnumValueDefinition: [:name],
+      InputObjectTypeDefinition: [:name, :fields],
+      TypeExtensionDefinition: [:definition]
+    }
```

Comment:
It overlaps in some way with the things that will fill up in `GraphQL.Type.*` here https://github.com/joshprice/graphql-elixir/blob/master/lib/graphql/type/definition.ex